src/merkle/simple_smt/mod.rs

```diff
@@ -182,6 +182,14 @@ impl SimpleSmt {
         self.get_path(index)
     }
 
+    // ITERATORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns an iterator over the leaves of this [SimpleSmt].
+    pub fn leaves(&self) -> impl Iterator<Item = (&u64, &Word)> {
+        self.leaves.iter()
+    }
```

Comment:
We should avoid using `&u64` as the pointer to a number should just be the number itself (i.e. often a pointer is a `u64`, so it is redundant to have `&u64` - unless it is `&mut u64` and we intend to change the value)

One suggestion is to do as follows:

```rust
pub fn leaves(&self) -> impl Iterator<Item = (&u64, &Word)> {
    self.leaves.iter().map(|(i, w)| (*i, w))
}
```

Reply:
I like this approach. I was a bit concerned about the extra overhead of `map` - but it probably should be negligible in this case.